web/src/user_status_ui.ts

```diff
@@ -185,35 +303,90 @@ function user_status_post_render(): void {
     });
 }
 
+function render_scheduled_end_time_widget(): void {
+    const tippy_props: Partial<tippy.Props> = {
+        placement: "bottom-start",
+    };
+    const opts: dropdown_widget.DropdownWidgetOptions = {
+        widget_name: "scheduled_end_time",
+        get_options: get_scheduled_end_time_widget_options,
+        item_click_callback: scheduled_end_time_widget_callback,
+        $events_container: $("#set-user-status-modal"),
+        tippy_props,
+        hide_search_box: true,
+        default_id: "never",
+        unique_id_type: "string",
+    };
+    scheduled_end_time_widget = new dropdown_widget.DropdownWidget(opts);
+    scheduled_end_time_widget.setup();
+}
+
+function scheduled_end_time_widget_callback(
+    event: JQuery.ClickEvent,
+    dropdown: tippy.Instance,
+): void {
+    event.preventDefault();
+    event.stopPropagation();
+    if (scheduled_end_time_widget.current_value === "custom") {
+        const now = new Date();
+        now.setSeconds(0, 0);
+        flatpickr.show_flatpickr(
+            $(".dropdown-list-item-common-styles")[4]!,
```

Comment:
Looks weird to select the element for flatpickr like this by using index in array of elements. Can we use the dropdown widget button as the element itself here?